homeassistant/components/klyqa/datacoordinator.py

```diff
@@ -0,0 +1,224 @@
+        args_in: Any,
+        timeout_ms: Any = 5000,
+        async_answer_callback: Any = None,
+    ) -> Any:
+        """Send_to_bulbs."""
+        ret = await super()._send_to_bulbs(
+            args_parsed,
+            args_in,
+            self.udp,
+            self.tcp,
+            timeout_ms=timeout_ms,
+            async_answer_callback=async_answer_callback,
+        )
+        return ret
+
+    # pylint: disable=arguments-differ
+    async def login(self, **kwargs: Any) -> bool:
+        """Login."""
+        ret: bool = await super().login(**kwargs)
+        if ret:
+            await api.async_json_cache(
+                {
+                    CONF_USERNAME: self.username,
+                    CONF_PASSWORD: self.password,
+                    CONF_SCAN_INTERVAL: self.scan_interval_conf,
+                    CONF_SYNC_ROOMS: self.sync_rooms,
+                    CONF_POLLING: self.polling,
+                    CONF_HOST: self.host,
+                },
+                "last.klyqa_integration_data.cache.json",
+            )
+        return ret
+
+    async def update_account(self) -> bool:
+        """Update_account."""
+
+        await self.request_account_settings()
+        if (
+            self.hass
+            and EVENT_KLYQA_NEW_LIGHT
+            in self.hass.bus._listeners  # pylint: disable=protected-access
+        ):
+            ha_entities = self.hass.data["light"].entities
+
+            for device in self.acc_settings["devices"]:
+                # look if any onboarded device is not in the entity registry already
+                u_id = api.format_uid(device["localDeviceId"])
+
+                light = [
+                    entity
+                    for entity in ha_entities
+                    if hasattr(entity, "u_id") and entity.u_id == u_id
+                ]
+
+                if len(light) == 0:
+                    if device["productId"].startswith("@klyqa.lighting"):
+                        # found klyqa device not in the light entities
+                        self.hass.bus.fire(EVENT_KLYQA_NEW_LIGHT, device)
+
+            for group in self.acc_settings["deviceGroups"]:
+                u_id = api.format_uid(group["id"])
+
+                light = [
+                    entity
+                    for entity in ha_entities
+                    if hasattr(entity, "u_id") and entity.u_id == u_id
+                ]
+
+                if len(light) == 0:
+                    # found klyqa device not in the light entities
+                    if (
+                        len(group["devices"]) > 0
+                        and "productId" in group["devices"][0]
+                        and group["devices"][0]["productId"].startswith(
+                            "@klyqa.lighting"
+                        )
+                    ):
+                        self.hass.bus.fire(EVENT_KLYQA_NEW_LIGHT_GROUP, group)
+        return True
+
+
+class KlyqaDataCoordinator(EntityComponent):
```

Comment:
Coordinator should not extend entity component (nothing should really).

Please extend the `DataUpdateCoordinator` instead.

Reply:
Could I stay without a DataUpdateCoordinator or is it a must for the merge.